test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -524,6 +560,163 @@ protected static IndexReader maybeWrapReaderEs(DirectoryReader reader) throws IO
+                fail();
+            }
+            MappedFieldType fieldType = mapper.fieldType();
+
+
+            try (Directory directory = newDirectory()) {
+                RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
+                writeTestDoc(fieldType, fieldName, indexWriter);
+                indexWriter.close();
+
+                try (IndexReader indexReader = DirectoryReader.open(directory)) {
+                    IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+                    AggregationBuilder aggregationBuilder = createAggBuilderForTypeTest(fieldType, fieldName);
+
+                    try {
+                        searchAndReduce(indexSearcher, new MatchAllDocsQuery(), aggregationBuilder, fieldType);
+                        if (supportedVSTypes.contains(fieldType.getValuesSourceType()) == false) {
+                            fail("Aggregator [" + aggregationBuilder.getType() + "] should not support field type ["
+                                + fieldType.typeName() + "] but executing against the field did not throw an excetion");
+                        }
+                    } catch (Exception e) {
+                        if (supportedVSTypes.contains(fieldType.getValuesSourceType())) {
+                            fail("Aggregator [" + aggregationBuilder.getType() + "] supports field type ["
+                                + fieldType.typeName() + "] but executing against the field threw an exception: [" + e.getMessage() + "]");
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void writeTestDoc(MappedFieldType fieldType, String fieldName, RandomIndexWriter iw) throws IOException {
+
+        if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.NUMERIC)) {
+            iw.addDocument(singleton(new SortedNumericDocValuesField(fieldName, randomLong())));
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.DATE)) {
+            iw.addDocument(singleton(new SortedNumericDocValuesField(fieldName, randomNonNegativeLong())));
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.BYTES)) {
+            if (fieldType.typeName().equals(BinaryFieldMapper.CONTENT_TYPE)) {
+                iw.addDocument(singleton(new BinaryFieldMapper.CustomBinaryDocValuesField(fieldName, new BytesRef("a").bytes)));
+            } else {
+                iw.addDocument(singleton(new SortedSetDocValuesField(fieldName, new BytesRef("a"))));
+            }
+        } else if (fieldType.getValuesSourceType().equals(CoreValuesSourceType.RANGE)) {
+            Object start;
+            Object end;
+            RangeType rangeType;
+
+            if (fieldType.typeName().equals(RangeType.DOUBLE.typeName())) {
+                start = randomDouble();
+                end = randomDoubleBetween(Math.nextUp((Double)start), Double.MAX_VALUE, false);
```

Comment:
I'm concerned we might introduce some instability by trying to aggregate over arbitrarily large ranges, which isn't really supported (i.e. it eventually trips the circuit breaker).  I'd suggest just just setting the end value to the next value after the start.  Also, you can use `RangeType.nextUp(Object)` for all cases, it's always well defined.

Reply:
Ah nice, didn't realize the RangeType's had a `nextUp()` on them :+1: